node/p2p/blossomsub.go

```diff
@@ -936,28 +937,33 @@ func discoverPeers(
 			return
 		}
 
+		wg := &sync.WaitGroup{}
+		defer wg.Wait()
 		for peer := range peerChan {
 			peer := peer
-			if peer.ID == h.ID() ||
-				h.Network().Connectedness(peer.ID) == network.Connected ||
-				h.Network().Connectedness(peer.ID) == network.Limited {
-				return
```

Comment:
The `return` here basically stops the lookup process virtually instantly, but it does not correctly stop the `FindPeers` background goroutine, and that causes a leak.

Reply:
Nice catch, thanks!